odoo/http.py

```diff
@@ -1872,7 +1872,11 @@ def _set_request_dispatcher(self, rule):
                 for disp in _dispatchers.values()
                 if disp.is_compatible_with(self)
             ]
-            raise BadRequest(f"Request inferred type is compatible with {compatible_dispatchers} but {routing['routes'][0]!r} is type={routing['type']!r}.")
+            e =(f"Request inferred type is compatible with {compatible_dispatchers} "
+                f"but {routing['routes'][0]!r} is type={routing['type']!r}.")
+            res = UnsupportedMediaType(e).get_response()
+            res.headers['Accept'] = ', '.join(dispatcher_cls.mimetypes)
```

Comment:
Looks ugly, but it's correct. Just add a comment why we need to create the exception twice.